tests/NavigationTests.elm

```diff
@@ -0,0 +1,156 @@
+module NavigationTests exposing (..)
+
+import Expect
+import Html exposing (..)
+import Navigation
+import Test exposing (..)
+import TestContext exposing (..)
+
+
+type Msg
+    = UrlChange Navigation.Location
+    | PushUrl String
+    | ModifyUrl String
+    | Back Int
+    | Forward Int
+    | Load String
+    | Reload
+
+
+type alias Model =
+    { location : Navigation.Location, msgs : List Msg }
+
+
+init : Navigation.Location -> ( Model, Cmd Msg )
+init location =
+    ( { location = location, msgs = [] }, Cmd.none )
+
+
+initWithStringFlags : String -> Navigation.Location -> ( Model, Cmd Msg )
+initWithStringFlags flags location =
+    init location
+
+
+programUpdate : Msg -> Model -> ( Model, Cmd Msg )
+programUpdate msg model =
+    case msg of
+        UrlChange location ->
+            ( { model | location = location, msgs = msg :: model.msgs }, Navigation.modifyUrl location.pathname )
+
+        PushUrl url ->
+            ( model, Navigation.newUrl url )
+
+        ModifyUrl url ->
+            ( model, Navigation.modifyUrl url )
+
+        Back amount ->
+            ( model, Navigation.back amount )
+
+        Forward amount ->
+            ( model, Navigation.forward amount )
+
+        Load url ->
+            ( model, Navigation.load url )
+
+        Reload ->
+            ( model, Navigation.reload )
+
+
+sampleProgram : TestContext Model Msg
+sampleProgram =
+    Navigation.program UrlChange
+        { init = init
+        , update = programUpdate
+        , subscriptions = \_ -> Sub.none
+        , view = always <| div [] []
+        }
+        |> TestContext.start
+
+
+sampleProgramWithFlags : TestContext Model Msg
+sampleProgramWithFlags =
+    Navigation.programWithFlags UrlChange
+        { init = initWithStringFlags
+        , update = programUpdate
+        , subscriptions = \_ -> Sub.none
+        , view = always <| div [] []
+        }
+        |> TestContext.startWithFlags "foo"
+
+
+all : Test
+all =
+    describe "Navigation TestContext"
+        [ test "starting a navigation program" <|
+            \() ->
+                sampleProgram
+                    |> expectHref "https://elm.testable/"
```

Comment:
Yeah, I commented above about this, but from seeing the tests, I definitely think we should add `TestContext.startWithLocation` and `TestContext.startWithFlagsAndLocation` instead of hardcoding a default URL, if that's possible.

Reply:
Done, I believe taking a string is easier for the user to use